msl/equipment/resources/isotech/millik.py

```diff
@@ -0,0 +1,183 @@
+
+        :param channel: The channel to configure for resistance measurement.
+        :param meas_range: The measurement range in ohms. Selects from 115 Ohms, 460 Ohms and 500 kOhms.
+        :param norm: The sense current to use for measurement. Defaults to use normal (1 mA) sense current,
+            unless False in which case it uses root2*1 mA to determine self-heating effects.
+            Thermistors always use 2 μA.
+        :param fourwire: The wiring arrangement eg 3 or 4 wire.
+        :return: Returns the internal values set for range, current, and wiring.
+        """
+        if channel not in self.channel_numbers:
+            self.raise_exception(f"Channel {channel} is not available in the current measurement setup")
+        current = 'NORM' if norm else 'ROOT2'
+        wire = 4 if fourwire else 3
+        message = f'sens:chan {channel};sens:res:rang {meas_range};sens:curr {current};sens:res:wir {wire}'
+        self.write(message)
+
+        channel_set = self.query('sens:chan?')
+        assert int(channel_set) == channel
+
+        range_set = self.query('sens:res:rang?')
+        if range_set == 'Error: Invalid range':
+            self.raise_exception(range_set)
+        assert float(range_set) >= float(meas_range)
+
+        current_set = float(self.query('sens:curr?'))
+
+        wire_set = int(self.query('sens:res:wir?'))
+        assert wire_set == wire
+
+        self.log_info(f'milliK Channel {channel} set to use {range_set} Ohms, {current_set} A, and {wire_set}-wire configuration')
+
+        self.channel_configuration[channel] = ['resistance', meas_range, current, wire]
+
+    def read_channel(self, channel: int, n: int = 1) -> float | list[float]:
+        """Initiate and report a measurement using the conditions defined by :meth:`.configure_resistance_measurement`.
+
+        :param channel: The channel to read.
+        :param n: The number of readings to make.
+        :return: A list of n readings, or a single float value if only one reading is requested.
+        """
+        if channel not in self.channel_configuration:
+            self.raise_exception(f"Please first configure channel {channel} before attempting to read values")
+
+        # TODO assuming resistance for now; voltage or current measurement will need a different call.
+        mode, meas_range, current, wire = self.channel_configuration[channel]
+        assert mode == 'resistance'
+
+        readings = [float(self.query(f'meas:res{channel}? {meas_range},{current}, {wire}')) for _ in range(n)]
+
+        if len(readings) == 1:
+            return readings[0]
+
+        return readings
+
+    def read_all_channels(self, n: int = 1) -> list[float]:
+        """Read from all configured channels using the conditions defined by :meth:`.configure_resistance_measurement`.
+
+        :param n: The number of readings to average for each returned value.
+        :return: A list of values from all configured channels (in ascending order).
```

Comment:
Do you think it would be helpful to instead return the channel number along with the reading? I'm assuming the user knows which channels they're reading here...

Reply:
Yes, I do think that would be helpful.